packages/SwingSet/docs/delivery.md

```diff
@@ -809,33 +810,35 @@ The `subject` argument of `dispatch.notify()` specifies which Promise is
 being resolved. It is always translated into a `PromiseID`.
 
 `dispatch.notify()` will be sent to all Vats which had subscribed to hear
-about the resolution of that Promise. The Decider vat for a Promise will not
-generally subscribe themselves, since they are the ones causing the Promise
-to become resolved, so they have no need to hear about it from the kernel.
+about the resolution of that Promise [TODO: document ordering
+relevance/constraints]. The Decider vat for a Promise will not generally
+subscribe themselves, since they are the ones causing the Promise to become
+resolved, so they have no need to hear about it from the kernel.
 
-The `Resolution` value of `dispatch.notify()` may contain slots, which are
+The `Resolution` values of `dispatch.notify()` may contain slots, which are
 translated just like `Message.args`.
 
 TODO: After a Vat receives notice of a Promise being resolved, we might
-choose to remove that promise from the vat's c-list, and forbid the Vat from
+choose to remove that promise from the vat's C-List, and forbid the Vat from
```

Comment:
Let's change this paragraph: we do indeed remove the promise from the c-list ("c-list promise entries are retired upon resolution")

We also use the word "retired" to describe removing c-list *object* entries when one side no longer needs to talk about the object, which is somewhat after the "drop" that one side does when it can no longer *reach* that object, because of WeakMaps. So let's be sure to use "retire" for promises here, and not "drop".

We don't exactly forbid the vat from mentioning the `vpid` again (we don't keep a history of retired IDs), but if it did, the kernel would treat it as a brand new promise. Liveslots won't, because it uses a persistent incrementing counter to generate IDs, so it will never re-generate the old one.. oh, huh, ok that's a problem for vat-upgrade, I'll make a ticket #4730 about that.